### DISCUSSION TITLE
Upgrade path for @relation from preview to beta

### DISCUSSION THREAD

**Discussion by brandonweiss** [prisma/prisma#2048] [Q&A] [answered]
I just read through the release notes about the new @relation syntax. While the more complicated syntax is a bit of a bummer, I can understand why it’s probably needed. But what I’m not clear on is how to upgrade to it?

If I have this:

```prisma
model User {
  id        Int       @id @default(autoincrement())
  profile   Profile
}

model Profile {
  id      Int    @id @default(autoincrement())
  user    User
}
```

And change it to this:

```diff
model User {
  id        Int       @id @default(autoincrement())
  profile   Profile
}

model Profile {
  id      Int    @id @default(autoincrement())
+ user    User   @relation(fields: [userId], references: [id])
+ userId  Int    // relation scalar field (used in the `@relation` attribute above)
}
```

If I run `prisma save`/`prisma migrate` will it… somehow figure out how to migrate things properly? Or do I have to do all of these migrations manually myself? 😓 And if I do it manually, how do I get things to “re-sync” with the schema and the database? 🤔

**Comment by pantharshit00** [comment]
Hi,

If you are an existing preview user, you can run `prisma introspect` to generate a compatible datamodel from your database.

There is also an issue which looks into simplifying the syntax for some common scenarios: https://github.com/prisma/prisma/issues/2018

**Comment by nikolasburk** [comment, marked as answer]
Hey Brandon 👋 I've added a new section to the top of the [release notes](https://github.com/prisma/prisma/releases/tag/2.0.0-beta.1) that explains the concrete upgrade path for the relation syntax!

Basically you don't need to touch the DB and can manually adjust your Prisma schema like so:

```diff
model User {
  id      Int      @id @default(autoincrement())
  profile Profile?
}

model Profile {
  id      Int     @id @default(autoincrement())
  user    User    @relation(fields: [userId], references: [id])
+ userId  Int     @map("user") // relation scalar field (used in the `@relation` attribute above)
}

model Post {
  id         Int         @id @default(autoincrement())
  author     User        @relation(fields: [authorId], references: [id])
+ authorId   Int         @map("author") // relation scalar field  (used in the `@relation` attribute above)
}
```

**Reply by nikolasburk** [reply]
@brandonweiss please let me know if that resolves your issue (and also please mark this as an answer here so it's clear this question is resolved 🙂 )

**Reply by brandonweiss** [reply]
Thanks so much! 👌

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
